### PR TITLE
Read original app.config and add to it

### PR DESCRIPTION
**Pull request by pakrym**
@davidfowl

### CODE REVIEW

**Comment by pakrym** [comment]
Damn, didn't see your changes (

File: src/Microsoft.DotNet.Tools.Compiler/Program.cs [hunk 1]
```diff
@@ -507,13 +507,18 @@ private static void MakeRunnable(ProjectContext runtimeContext, string outputPat
 
         private static void GenerateBindingRedirects(ProjectContext runtimeContext, string outputPath, LibraryExporter exporter)
         {
+            var baseAppConfigPath = Path.Combine(runtimeContext.ProjectDirectory, "app.config");
```
Comment (davidfowl) [inline src/Microsoft.DotNet.Tools.Compiler/Program.cs]: Casing will break on *nix, we should probably support `App.config` also.

File: src/Microsoft.DotNet.Tools.Compiler/Program.cs [hunk 2]
```diff
@@ -507,13 +507,18 @@ private static void MakeRunnable(ProjectContext runtimeContext, string outputPat
...
+            XDocument baseAppConfig = null;
+            if (File.Exists(baseAppConfigPath))
+            {
+                baseAppConfig = XDocument.Load(File.OpenRead(baseAppConfigPath));
```
Comment (davidfowl) [inline src/Microsoft.DotNet.Tools.Compiler/Program.cs]: No dispose?

**Comment by davidfowl** [comment]
This code needs tests.

File: src/Microsoft.DotNet.Tools.Compiler/BindingRedirectGenerator.cs [hunk 3]
```diff
@@ -34,43 +34,77 @@ internal class BindingRedirectGenerator
 
         private readonly SHA1 _sha1 = SHA1.Create();
 
-        public XDocument Generate(IEnumerable<LibraryExport> dependencies)
+        public XDocument Generate(IEnumerable<LibraryExport> dependencies, XDocument document)
         {
-            var redirects = CollectRedirects(dependencies);
-            
+			var redirects = CollectRedirects(dependencies);
```
Comment (davidfowl) [inline src/Microsoft.DotNet.Tools.Compiler/BindingRedirectGenerator.cs]: Broken formatting

File: src/Microsoft.DotNet.Tools.Compiler/BindingRedirectGenerator.cs [hunk 4]
```diff
@@ -34,43 +34,77 @@ internal class BindingRedirectGenerator
...
-                            redirects.Select(GetDependentAssembly)
-                            )
-                        )
-                    )
-                );
             return document;
         }
 
-        private XElement GetDependentAssembly(AssemblyRedirect redirect)
+        private void AddDependentAssembly(AssemblyRedirect redirect, XElement assemblyBindings)
```
Comment (davidfowl) [inline src/Microsoft.DotNet.Tools.Compiler/BindingRedirectGenerator.cs]: We should clear out the existing binding redirects and just write new ones. I'm not sure we care about merging them.

Comment (pakrym) [inline src/Microsoft.DotNet.Tools.Compiler/BindingRedirectGenerator.cs]: Let's keep as much of user work as we can

Comment (davidfowl) [inline src/Microsoft.DotNet.Tools.Compiler/BindingRedirectGenerator.cs]: Except when we know it's wrong.

Comment (pakrym) [inline src/Microsoft.DotNet.Tools.Compiler/BindingRedirectGenerator.cs]: Is there what if user dynamically load another assembly which requires redirect but we cannot know about it?

Comment (davidfowl) [inline src/Microsoft.DotNet.Tools.Compiler/BindingRedirectGenerator.cs]: Is the assumption that there's no overlap?

**Comment by pakrym** [comment]
Works.

**Pull request merged**